File: java/src/main/java/sammancoaching/MidLevel.java
package sammancoaching;

class MidLevel implements SeniorityLevel {
    @Override
    public double getPensionContributionBonus(SalaryContributionPercentages databaseAccess) {
        return databaseAccess.lookupValue(SalaryContributionPercentages.MID_SENIORITY_PERCENTAGE);
    }
}

File: typescript/src/MidLevel.ts
import {SeniorityLevel} from "./SeniorityLevel";
import {ContributionPercentages, SalaryContributionPercentages} from "./SalaryContributionPercentages";

export class MidLevel implements SeniorityLevel {
  getPensionContributionBonus(databaseAccess: ContributionPercentages): number {
    return databaseAccess.lookupValue(SalaryContributionPercentages.MID_SENIORITY_PERCENTAGE);
  }
}
